telegram/error.py

```diff
@@ -97,6 +109,9 @@ def __init__(self, retry_after):
         super().__init__('Flood control exceeded. Retry in {} seconds'.format(retry_after))
         self.retry_after = float(retry_after)
 
+    def __reduce__(self):
+        return self.__class__, (self.retry_after,), {"message": self.message}
```

Comment:
@Bibo-Joshi This one is a bit tricky. The first time we initialize a `RetryAfter(12)` exception, the message will display the delay as an integer: `"Retry in 12 seconds"`. If `pickle` tries to re-build the exception, `self.retry_after` being a `float` the message will be changed to `"Retry in 12.0 seconds"`, causing the `assert error.message == unpickled.message` to fail. That's why I specifically update the `message` attribute.

Reply:
Mh, that's because retry_after is converted to float only after the message is generated. I think there's no harm in  converting before.